subsys/testsuite/ztest/include/ztest_main_if.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2019 Oticon A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SUBSYS_TESTSUITE_ZTEST_INCLUDE_ZTEST_MAIN_IF_H
+#define SUBSYS_TESTSUITE_ZTEST_INCLUDE_ZTEST_MAIN_IF_H
```

Comment:
Why don't you name it `ztest_unit.h` or `ztest_unit_main.h` ? It seems more coherent.

Reply:
Actually I'm not getting the purpose of this commit, even for unit test one still need to include `ztest.h` because there may be some required definitions there, right ? Even if not, you are still building `ztest.c` for unit tests and `ztest.h` is still including this header  ....

It seems to me that this is just bringing more complication ... but I may be wrong here.

Reply:
The purpose of this, is being able to compile ztest_unit_main.c without pulling all other zephyr headers into that translation unit. 
This ztest_unit_main defines the Linux executable `main()`. Meaning, it is just part of the "unit tests framework".
Compiling it with other zephyr headers as it was, complicates things quite a lot unnecessarily. Separating it, is the easiest way of having the new definition this PR introduces not create problems in the unit test framework. It should also avoid other intermingling problems in the future.
Wrt. this new header in particular, it is just what this `main()` requires from the rest of the unit test framework.